docs/developers/frontend-shared.rst

```diff
@@ -0,0 +1,67 @@
+=========================
+Frontend Shared Package
+=========================
+
+The frontend-shared package is located in a sub directory called ``./frontend-shared`` inside this
+repository. The purpose of this package is to serve as a common library for shared utilities,
+components and styles that span across our frontend applications.
+
+The long term goal is to move this package into its own repository, but in the short term it
+lives in the client as a convenience to leverage the existing tooling. This also gives developers
+the benefit of working with just one repository when making changes that mostly affect the client
+in the first place.
+
+How it works
+=========================
+
+Frontend-shared works by creating a symlink via ``yarn link`` between the working dir
+``./frontend-shared/`` and the installed dir ``node_modules/@hypothesis/frontend-shared/``.
+
+Client setup
+=========================
+
+To setup frontend-shared in the client, simply run
+
+``$ make dev``
+
+Internally, that will trigger the frontend-shared setup which transpiles all of the jsx and
+outputs that result to ``frontend-shared/lib`` folder. The next thing it does is creates a linked
+package in the client ``node_modules/@hypothesis`` folder which points to ``frontend-shared/lib``.
+When running `make dev` this pipeline is automated and will trigger on file changes. These
+steps can also be run directly too via
```

Comment:
It took me a while to grok the sentence about running the build manually! :)

```suggestion
This will trigger the ``frontend-shared`` build which:

 * builds transpiled JSX into the ``frontend-shared/lib`` directory
 * creates a linked package in the ``node_modules/@hypothesis`` folder (this is linked to ``frontend-shared/lib``)

When ``make dev`` is running, file changes will automatically trigger a rebuild. To rebuild manually:
```